content/blog/2024-06-18-gsoc-uefi-gop.mdx

```diff
@@ -0,0 +1,43 @@
+---
+title: "GSoC'24: UEFI Graphics Output Protocol Support in Unikraft"
+description: |
+  This project aims to add support for the modern UEFI GOP interface in Unikraft.
+publishedDate: 2024-06-18
+authors:
+- Sriprad Potukuchi
+tags:
+- gsoc
+- gsoc24
+- uefi
+- booting
+---
+
+## Project Overview
+<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-generic-badge.svg" align="right" />
+
+Unikraft's ability to print logs to the screen is quite limited, which makes it harder to debug it during development. While it is entirely possible to use the serial port console, it is not as easily accessible or available on usual consumer machines.
+
+The widely available and standardized [UEFI Graphics Output Protocol](https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html#efi-graphics-output-protocol) (GOP) interface is an excellent candidate. It provides a raw framebuffer, which can be used to set individual pixels on the screen.
+
+This project aims to implement a console based on UEFI GOP, which is capable of parsing fonts and displaying text on the screen by setting the appropriate pixels in the framebuffer.
```

Comment:
Can you also describe how this framebuffer is fetched, what it actually is, its state after ExitBootServices is called, etc.

Reply:
I've written more about the framebuffer. Couldn't fit info about ExitBootServices into the flow though. Should I work on providing more context here?

Reply:
Update: I've brought in some context and talked about the aforementioned points :D